### PR TITLE
Most queens v2 oop

### PR DESCRIPTION
**Pull request by bennyrubanov**

### CODE REVIEW

File: src/metrics/promotions.ts [hunk 1]
```diff
@@ -33,21 +35,62 @@ export class PromotionMetric implements Metric {
...
+
+      // identify the maxQueenCount in a particular move, and the games and moves that the maxQueenCount occured in
+      // push to array of games and moves if tie, otherwise wipe the array and add new game
+      // only add one move entry for each game maxQueenCount (rather than one entry for each move that the maxQueenCount appears in)
+      if (thisGameQueenCount > this.maxQueenCounts) {
+        this.maxQueenCounts = thisGameQueenCount;
+        const gameSite = metadata.find((item) => item.startsWith('[Site "'))
+          ?.replace('[Site "', '')
+          ?.replace('"]', '');
+        const moveOS = move.originalString;
```
Comment (EllAchE) [inline src/metrics/promotions.ts:76]:
```suggestion
```
Just directly use the move object

File: src/metrics/promotions.ts [hunk 2]
```diff
@@ -33,21 +35,62 @@ export class PromotionMetric implements Metric {
...
+        if (!this.movesAndGamesWithMaxQueenCount.some((item) => item.game === gameSite)) {
+          this.movesAndGamesWithMaxQueenCount = [{
+            game: gameSite,
+            move: moveOS
+          }];
+        }
+      } else if (thisGameQueenCount === this.maxQueenCounts) {
```
Comment (EllAchE) [inline src/metrics/promotions.ts:83]: there are bugs with this logic. You could solve them by using a set instead. But as it is if a game goes up to the max, loses a queen then gets another you'll have the same game 2x

File: src/metrics/promotions.ts [hunk 3]
```diff
@@ -33,21 +35,62 @@ export class PromotionMetric implements Metric {
...
         // update maps
         this.promotionMap[move.uas][move.promotion]++;
+        
+        // increment queen count if a promotion occurs
+        if (move.promotion === "q") {
+          thisGameQueenCount++;
+        }
       }
 
-      for (const piece of board) {
```
Comment (EllAchE) [inline src/metrics/promotions.ts:50]: was this just unused loop?

File: src/metrics/promotions.ts [hunk 4]
```diff
@@ -33,21 +35,62 @@ export class PromotionMetric implements Metric {
...
 
       if (move.originalString.includes('=')) {
         // update maps
         this.promotionMap[move.uas][move.promotion]++;
+        
+        // increment queen count if a promotion occurs
+        if (move.promotion === "q") {
+          thisGameQueenCount++;
+        }
       }
```
Comment (EllAchE) [inline src/metrics/promotions.ts:61]: minor change but we can directly use the .promotion flag's existence

**Review by EllAchE — APPROVED** [review]
Approving but there is a bug

File: src/metrics/promotions.ts (hunk 3 above)
Comment (bennyrubanov) [reply src/metrics/promotions.ts:50]: yes - thought i got rid

File: src/metrics/promotions.ts (hunk 4 above)
Comment (bennyrubanov) [reply src/metrics/promotions.ts:61]: wdym?

File: src/metrics/promotions.ts (hunk 2 above)
Comment (bennyrubanov) [reply src/metrics/promotions.ts:83]: I disagree. There is a check to see if the game already exists in the movesAndGamesWithMaxQueenCount array. If the game exists in that array it won't be added

File: src/metrics/promotions.ts (hunk 3 above)
Comment (bennyrubanov) [reply src/metrics/promotions.ts:50]: did you remove already? i don't see it in the code

**Pull request merged**